cfy_manager/components/base_component/base_component.py

```diff
@@ -82,3 +88,90 @@ def validate_dependencies(self):
+
+    def use_supplied_certificates(self, cert_destination, key_destination,
+                                  ca_destination,
+                                  owner=CLOUDIFY_USER, group=CLOUDIFY_GROUP):
+        """
+            Use user-supplied certificates, after checking they're not
+            broken.
+        """
+        cert_src, key_src, ca_src, key_pass = check_certificates(
+            self.component_name,
+        )
+
+        if not any([cert_src, key_src, ca_src, key_pass]):
+            # No certificates supplied, so not using them
+            self.logger.debug('No user-supplied certificates were present.')
+            return False
+
+        if key_pass:
+            # TODO: Strip password from key
+            raise ValidationError('Key must not have password.')
+
+        # Put the files in the correct place
+        self.logger.info('Ensuring files are in correct locations.')
+        for path in cert_destination, key_destination, ca_destination:
+            directory = os.path.dirname(path)
+            if not os.path.exists(directory):
+                self.logger.info(
+                    'Ensuring certificate destination {directory} exists '
+                    'and is accessible.'.format(directory=directory)
+                )
+                sudo(['mkdir', '-p', directory])
+                sudo(['chmod', 'o+x', directory])
+            if not os.path.isdir(directory):
+                raise ValidationError(
+                    'Certificate destination directory {directory} exists '
+                    'but is not a directory!'.format(directory=directory)
+                )
+
+        if cert_src != cert_destination:
+            self.logger.info(
+                'Moving cert from {src} to {dest}'.format(
+                    src=cert_src,
+                    dest=cert_destination,
+                )
+            )
+            sudo(['mv', cert_src, cert_destination])
+        if key_src != key_destination:
+            self.logger.info(
+                'Moving key from {src} to {dest}'.format(
+                    src=key_src,
+                    dest=key_destination,
+                )
+            )
+            sudo(['mv', key_src, key_destination])
+        if ca_src != ca_destination:
+            if ca_src:
+                self.logger.info(
+                    'Moving CA cert from {src} to {dest}'.format(
+                        src=ca_src,
+                        dest=ca_destination,
+                    )
+                )
+                sudo(['mv', ca_src, ca_destination])
+            else:
+                self.logger.info(
+                    'Self signed cert, copying cert to {dest}'.format(
+                        dest=ca_destination,
+                    )
+                )
+                sudo(['cp', cert_destination, ca_destination])
+
+        self.logger.info('Setting certificate ownership and permissions.')
+        sudo(['chown', owner + '.' + group,
```

Comment:
We usually use `'{0}.format` as in the rest of the file instead of concatenation. It's probably better to keep it with one standard.

Reply:
Can do- for this bit I actually felt like the concat version was more readable, but updated.